app/src/features/projects/view/components/Species.tsx

```diff
@@ -37,21 +49,66 @@ const useStyles = makeStyles({
 const Species: React.FC<ISpeciesProps> = (props) => {
   const {
     projectForViewData: {
-      species: { focal_species, ancillary_species }
-    }
+      species: { focal_species, ancillary_species },
+      id
+    },
+    codes
   } = props;
 
+  const history = useHistory();
+  const biohubApi = useBiohubApi();
+
   const classes = useStyles();
 
+  const [openEditDialog, setOpenEditDialog] = useState(false);
+  const [openErrorDialog, setOpenErrorDialog] = useState(false);
```

Comment:
I would change this to be consistent with how we are probably going to do it in the final version:

```
const [errorDialogProps, setErrorDialogProps] = useState<IErrorDialogProps>({
    dialogTitle: EditSpeciesI18N.editErrorTitle,
    dialogText: EditSpeciesI18N.editErrorText,
    open: false,
    onClose: () => {
      setErrorDialogProps({ ...errorDialogProps, open: false });
    },
    onOk: () => {
      setErrorDialogProps({ ...errorDialogProps, open: false });
    }
  });

  const showErrorDialog = (textDialogProps?: Partial<IErrorDialogProps>) => {
    setErrorDialogProps({ ...errorDialogProps, ...textDialogProps, open: true });
  };
```